peps/pep-9999.rst

```diff
@@ -0,0 +1,120 @@
+PEP: 9999
+Title: imath --- module for number-theoretic functions
+Author: Sergey B Kirpichev
+Sponsor: Victor Stinner <vstinner@python.org>
+Discussions-To: Pending
+Status: Draft
+Type: Standards Track
+Created: 10-May-2025
+Python-Version: 3.15
+Post-History: 02-Jun-2019,
+              `09-May-2025 <https://discuss.python.org/t/91337>`__,
+
+
+Abstract
+========
+
+This PEP describes new module for number-theoretic functions, like
+:external+py3.14:func:`math.gcd` or :external+py3.14:func:`math.isqrt`.
+
+
+Motivation
+==========
+
+The :external+py3.14:mod:`math` documentaton says: "This module provides
+access to the mathematical functions defined by the C standard."  But as a
+state of art, over time the module was populated with functions that aren't
+related to the C standard or floating-point arithmetics.  Now it's much
+harder to describe module scope, content and interfaces (e.g. returned
+values).
+
+Apparently, the :external+py3.14:mod:`math` can't serve as a catch-all place
+for mathematical functions: we have also the :external+py3.14:mod:`cmath` and
+the :external+py3.14:mod:`statistics`.  Lets make same for integer-related
+functions.
+
+
+Specification
+=============
+
+The PEP proposes moving the following integer-related functions [1]_ to a new
+module, called ``imath``:
+
+* :external+py3.14:func:`~math.gcd`
+* :external+py3.14:func:`~math.lcm`
+* :external+py3.14:func:`~math.isqrt`
+* :external+py3.14:func:`~math.factorial`
+* :external+py3.14:func:`~math.comb`
+
+Their aliases in :external+py3.14:mod:`math` will be :term:`soft deprecated`.
+
+Possible extensions for the new module and it's scope are discussed in the
+Open Issues section.
+
+
+Backwards Compatibility
+=======================
+
+As aliases in :external+py3.14:mod:`math` will be kept for indefinite
+time (their use would be discouraged), there are no anticipated code breaks.
+
+
+Reference Implementation
+========================
+
+TBD
+
+
+Open Issues
+===========
+
+Module name
+-----------
+
+Another variants for name: ``ntheory`` (like SymPy's submodule), ``intmath``,
+``integermath`` and ``imaths``.
+
+Possible renaming of isqrt
+--------------------------
+
+As there is no clash with floating-point version, it could be renamed to
+``sqrt`` (Java-like).
+
+Module scope and possible extensions
+------------------------------------
+
+Unless we can just provide bindings to some well supported mathematical
+library like the GMP, the module scope should be limited.  For example, no
+primality testing and factorization.
```

Comment:
Thank you :-) IMO primality testing is a can a worm and a bad idea for the stdlib :-(

Reply:
That was just a first example coming in my mind.

I think it's a good idea to show some border line.  For the math/cmath modules we have the C standard to reject new stuff.  Here it's less obvious.